src/cbor.h

```diff
@@ -212,7 +212,20 @@ typedef enum CborEncoderAppendType
     CborEncoderApendRawData = 2
 } CborEncoderAppendType;
 
-typedef CborError (*CborEncoderWriteFunction)(void *, const void *, size_t, CborEncoderAppendType);
+/**
```

Comment:
I prefer the documentation in the .c source file so the header is clean and small. Makes it easier to search for and find things in it, at a glance. Repeats.

Reply:
Ahh okay, I normally put docs in the headers out of habit because of packaging: when you ship a binary build, usually you ship a library object (`.a` / `.so` / `.dll`) and headers that describe the functions within; such users can't see comments in the C sources.

`tinycbor` being statically linked into the sources though and _not_ shipped separately, is a bit different.  I'll move the documentation comments. :-)

Reply:
Yeah, conflicting requirements. I prefer the comments in the sources because I usually read the .h itself. Of course, a good IDE can just collapse the comments...

In the olden days, that meant longer compile times for bigger source.